### PR TITLE
Adds some notes on commits and GitHub to CONTRIBUTING

### PR DESCRIPTION
**Pull request by sr-gi**
Some newcomers are struggling with our ways of doing things, so it may be best to document what we expect so there are less friction/back-and-forth conversations in reviews.

### CODE REVIEW

**Comment by sr-gi** [comment]
@mariocynicys would you mind adding anything you see fit?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,6 +72,60 @@ pub struct Responder {
...
 Tests should be provided to cover both positive and negative conditions. Tests should cover both the proper execution as well as all the covered error paths. PR with no proper test coverage will not be merged.
 
-## Signing Commits
+## Git conventions 
+
+### Commits, descriptions, and long descriptions
+
+- Changes must be split logically in commits, such that a commit is self-contained. 
+- In general terms, all commits need to pass the test suite. There may be some exceptions to this rule if the change you are working on touches several components of the codebase and it makes more sense to split the change by component (or group of components).
+- Commit descriptions need to be short and explanatory. If we are, for instance, adding an RPC command to the backend, "Adds command X to the backend" will be a good short description, "Add command" or "Fix #123" where #123 is an issue referencing this feature **IS NOT**.
```
Comment (mariocynicys) [inline CONTRIBUTING.md:81]: I think what's referred to as `Commit descriptions` here are actually called `Commit messages`.

**Comment by mariocynicys** [comment]
> @mariocynicys would you mind adding anything you see fit?

Nothing in mind tbh.
But I would say I think the last section added in this PR (`Remotes`) is somewhat off-topic. It's more of a general git info than a contribution guideline.

**Comment by sr-gi** [comment]
> > @mariocynicys would you mind adding anything you see fit?
> 
> Nothing in mind tbh. But I would say I think the last section added in this PR (`Remotes`) is somewhat off-topic. It's more of a general git info than a contribution guideline. People manage their remotes differently and there is already online resources on how to do the same.

I kinda agree. People were struggling a bit with how to work with branches and such so I decided to add it. But it may not be a good fit tbh

File: CONTRIBUTING.md (hunk 1 above)
Comment (sr-gi) [reply CONTRIBUTING.md:81]: I was checking some external sources and felt like there was no consensus on this, given both the short and long messages are called messages 😅 

I think we could settle it, conceptually, as Commit summaries (or titles) and commit descriptions: https://www.gitkraken.com/learn/git/best-practices/git-commit-message#git-commit-message-structure

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -72,6 +72,35 @@ pub struct Responder {
...
+	67ac345 Addresses review comments
+	7dc7fcd Updates X because G was missing
+	b60999c Adds missing test
+	...
+	```
+	
+- Create a new branch to work on your pull request. **DO NOT** work from the master branch of your fork*
+- **DO NOT** merge master into your branch, rebase master instead*
+
+* If you're not sure how to handle this, check external documentation on how to manage multiple remotes for the same repository.
```
Comment (mariocynicys) [inline CONTRIBUTING.md:102]: I guess you meant to mention this as a note for the two `*` references in the previous lines?
But it renders as another item in the list. Dunno why :/

If I'm right there, you can format it like this:
```
	<sup>* If you're not sure how to handle this, check external documentation on how to manage multiple remotes for the same repository.<sup>
```
Note: <sup> is just to make the note small in size.
Note: extra tap to indent it with the items in the list.

**Review by mariocynicys — APPROVED** [review]
LGTM

**Pull request merged**